broker-core/src/test/java/io/zeebe/broker/workflow/TimerStartEventTest.java

```diff
@@ -117,28 +119,43 @@ public void shouldCreateTimer() {
   public void shouldTriggerAndCreateWorkflowInstance() {
     // when
     final ExecuteCommandResponse response = testClient.deployWithResponse(SIMPLE_MODEL);
-    final DeploymentRecordValue deploymentRecord =
+    final DeployedWorkflow workflow =
         testClient
             .receiveFirstDeploymentEvent(DeploymentIntent.CREATED, response.getKey())
-            .getValue();
+            .getValue()
+            .getDeployedWorkflows()
+            .get(0);
 
     // then
     assertThat(RecordingExporter.timerRecords(TimerIntent.CREATED).exists()).isTrue();
     brokerRule.getClock().addTime(Duration.ofSeconds(2));
 
-    assertThat(RecordingExporter.timerRecords(TimerIntent.TRIGGERED).exists()).isTrue();
-    assertThat(RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.CREATE).exists())
-        .isTrue();
-
-    final WorkflowInstanceRecordValue wfInstanceRecord =
-        RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.ELEMENT_READY)
-            .getFirst()
-            .getValue();
-
-    Assertions.assertThat(wfInstanceRecord)
-        .hasVersion(1)
+    Assertions.assertThat(
+            RecordingExporter.workflowInstanceRecords(WorkflowInstanceIntent.EVENT_TRIGGERING)
+                .getFirst()
+                .getValue())
+        .hasElementId("start_1")
         .hasBpmnProcessId("process")
-        .hasWorkflowKey(deploymentRecord.getDeployedWorkflows().get(0).getWorkflowKey());
+        .hasVersion(workflow.getVersion())
+        .hasWorkflowKey(workflow.getWorkflowKey());
+
+    assertThat(
+            RecordingExporter.getRecords()
+                .stream()
+                .skip(2)
```

Comment:
Why skip 2? Why not simply start asserting after the `TimerIntent.TRIGGER` event? This is what the test is testing, no? I assume we have other tests testing the creation of the timer on deployment.

So we could find the first `TimerIntent.TRIGGER` then filter the stream so we only check records with position > than that, and test only what the test cares about.